utils/index.d.ts

```diff
@@ -1711,10 +1714,71 @@ export declare interface Wrapper {
     unwrap<T>(): T;
 }
 
+// temporary
+type ResourceGroupsItem = unknown;
+
 /**
  * Tests to see if something is a wrapper, by ensuring it is an object
  * and has an "unwrap" function
  * @param maybeWrapper An object to test if it is a wrapper
  * @returns True if a wrapper, false otherwise
  */
 export declare function isWrapper(maybeWrapper: unknown): maybeWrapper is Wrapper;
+
+export declare function appResourceExperience<TPick extends unknown>(context: IActionContext, tdp: TreeDataProvider<ResourceGroupsItem>, resourceTypes?: AzExtResourceType | AzExtResourceType[], childItemFilter?: ContextValueFilter): Promise<TPick>;
```

Comment:
Maybe namespacing could be helpful? E.g.:

```TypeScript
picks.byResourceType(context, tdp, resourceTypes, ....);
```
Not sure what wording would work best, but an idea.